src/main/java/pl/sda/climbing_shop/configuration/UserDetailsServiceImplForEmployees.java

```diff
@@ -0,0 +1,70 @@
+package pl.sda.climbing_shop.configuration;
+
+
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Component;
+import pl.sda.climbing_shop.employee.Employee;
+import pl.sda.climbing_shop.employee.EmployeeRepository;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@RequiredArgsConstructor
+@Component
+public class UserDetailsServiceImplForEmployees implements UserDetailsService {
+
+    private final EmployeeRepository employeeRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```

Comment:
może warto by ugenerycznić tą metode loadUserByUsername<T>  albo klase UserDetailsService <T>   (nie pamietam jak w javie to jest)
gdzie T moze byc employyeRepository lub customerRepository